### PR TITLE
Endpoint for admins to update user membership

### PR DESCRIPTION
**Pull request by jeffplays2005**
Created two endpoints, "promote" and "demote", both takes a PUT request. 
Checks for forbidden requests(is admin) and conflicting requests(already  promoted/demoted).
Updates UserService and AuthService accordingly.  
AuthService updated to remove claims when "null" is passed.

### CODE REVIEW

File: server/src/service-layer/controllers/UserController.ts [hunk 1]
```diff
@@ -75,4 +79,62 @@ export class UsersController extends Controller {
...
+    @Body() requestBody: promoteUserRequestBody
+  ): Promise<void> {
+    const userService = new UserDataService() // create a new data service
+    const authService = new AuthService()
+    // attempt to fetch user
+    const user: UserAdditionalInfo = await userService.getUserData(
+      requestBody.uid
+    )
+    if (!user) return this.setStatus(400) // bad request
+    if (user.membership === "admin") return this.setStatus(403) // admin forbidden
```
Comment (bcho892) [inline server/src/service-layer/controllers/UserController.ts:98]: we should not be checking this in firestore database, we need to check the `customClaims option` on the user. Should add a method in `AuthService` thats uses `auth.getUser(uid)`, the `userRecord` returned will have a `customClaims` property

**Review by bcho892 — COMMENTED** [review]
looks good, just need to not rely on firestore for user roles

File: server/src/service-layer/controllers/UserController.ts [hunk 2]
```diff
@@ -75,4 +79,62 @@ export class UsersController extends Controller {
...
+    if (user.membership === "member") return this.setStatus(409) // conflict
+    try {
+      // update user in UserService
+      await userService.editUserData(requestBody.uid, {
+        membership: "member"
```
Comment (bcho892) [inline server/src/service-layer/controllers/UserController.ts:103]: hm, we might have to update `UserAdditionalInfo` to allow `undefined`

File: server/src/business-layer/services/AuthService.test.ts
Comment (bcho892) [inline server/src/business-layer/services/AuthService.test.ts:1]: 👍

**Review by bcho892 — COMMENTED** [review]
LGTM, just need some tests

**Review by bcho892 — APPROVED** [review]
👍 i just cleaned up to avoid side effects by properly cleaning up after each test

**Comment by jeffplays2005** [comment]
thank you so much for the commits benson!

**Pull request merged**